version: '3.8'

services:
  postgres:
    image: postgres:17
    container_name: postgres-container
    ports:
      - "5432:5432"
    environment:
      POSTGRES_USER: acelera
      POSTGRES_PASSWORD: programador
      POSTGRES_DB: api_pedidos
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./docker/postgres/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - docker-network
    restart: always
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "acelera"]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: api-pedidos-container
    ports:
      - "8080:8080"
    environment:
      SERVER_PORT: 8080
      DB_URL: jdbc:postgresql://postgres-container:5432/api_pedidos
      DB_USERNAME: acelera
      DB_PASSWORD: programador
      BASE_URL_MULTT: https://api.multt.digital/api/pagamentos
      TOKEN_MULTT: 59c6d7b0-1180-4aeb-ae0a-c7d30f464a73
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - docker-network
    restart: always

volumes:
  postgres_data:

networks:
  docker-network:
    driver: bridge
